### PR TITLE
feat(glue): default data location for tables is the root of the bucket

### LINKED ISSUE #8472: Default data location for AWS Glue tables
### The Question
I started rolling out a Glue table with the following CDK structs:

```
const trailTable = new Table(this, 'TrailTable', {
    bucket: <ref to bucket>,
    database: <ref to db>,
    tableName: 'table',
    columns: [{
        name: 'user',
        type: Schema.STRING,
    }],
    dataFormat: DataFormat.JSON,
});
```

To my surprise the Glue table pointed towards my bucket using this URL: `s3://<bucket>/data`.

Looking into the [documentation](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-glue.Table.html#s3prefix-span-class-api-icon-api-icon-experimental-title-this-api-element-is-experimental-it-may-change-without-notice-span) of CDK, indeed `/data` is the default location for the Glue data to be discovered from (this is the `s3Prefix` property). But little explanation is given why this is the default. Is this done to follow certain guidelines or is this a randomly chosen path? 

I would have expected the default to be just empty; no nested folder in the bucket but just the root. Defining the blank `s3Prefix` seems to be out of place to achieve this behavior:

```
const trailTable = new Table(this, 'TrailTable', {
    ...,
    s3Prefix: '',
});
```

### Proposed solution

Remove the default that points to `/data` for the `s3Prefix` and use empty string instead

OR

Provide documentation that explains why `/data` is chosen as default

### PR DESCRIPTION
**Pull request by DerkSchooltink**
Fixes #8472

BREAKING CHANGE: The default location of glue data will be the root of an s3 bucket, instead of `/data`

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-glue/lib/table.ts [hunk 1]
```diff
@@ -87,7 +87,7 @@ export interface TableProps {
   /**
    * S3 prefix under which table objects are stored.
    *
-   * @default data/
+   * @default empty string
```
Comment (iliapolo) [inline packages/@aws-cdk/aws-glue/lib/table.ts:90]:
```suggestion
   * @default - No prefix. The data will be stored under the root of the bucket.
```

File: packages/@aws-cdk/aws-glue/lib/table.ts [hunk 2]
```diff
@@ -236,7 +236,7 @@ export class Table extends Resource implements ITable {
 
     this.database = props.database;
     this.dataFormat = props.dataFormat;
-    this.s3Prefix = (props.s3Prefix !== undefined && props.s3Prefix !== null) ? props.s3Prefix : 'data/';
+    this.s3Prefix = (props.s3Prefix !== undefined && props.s3Prefix !== null) ? props.s3Prefix : '';
```
Comment (iliapolo) [inline packages/@aws-cdk/aws-glue/lib/table.ts:239]:
```suggestion
    this.s3Prefix = props.s3Prefix ?? ''
```

File: packages/@aws-cdk/aws-glue/README.md [hunk 3]
```diff
@@ -55,6 +55,8 @@ new glue.Table(stack, 'MyTable', {
 });
 ```
 
+The default for `s3Prefix` is the root of the bucket.
```
Comment (iliapolo) [inline packages/@aws-cdk/aws-glue/README.md:58]: I suggest to combine this with the sentence just above the snippet:

```text
By default, an S3 bucket will be created to store the table's data and stored in the bucket root. You can also manually pass the `bucket` and `s3Prefix`:
```

**Comment by DerkSchooltink** [comment]
Just a sanity check: this introduces a breaking change for people relying on the default `s3Prefix` that points to `/data`. I've mentioned this in the description of this PR, is this sufficient communication to make clear something breaking has been introduced? @iliapolo

**Comment by iliapolo** [comment]
@DerkSchooltink Yes. The breaking change notice will appear in our release notes and changelog 👍

**Pull request merged**